django/utils/functional.py

```diff
@@ -20,12 +20,19 @@ class cached_property:
 
     Optional ``name`` argument allows you to make cached properties of other
     methods. (e.g.  url = cached_property(get_absolute_url, name='url') )
+
+    On Python 3.6 or more, the name argument isn't needed.
+    (e.g.  url = cached_property(get_absolute_url) )
     """
     def __init__(self, func, name=None):
         self.func = func
         self.__doc__ = getattr(func, '__doc__')
         self.name = name or func.__name__
```

Comment:
I think this should throw a warning on `< PY36` where `name.startswith('__')`

Reply:
Not sure about the best way to do that though

Reply:
I think a plain warning would be fine.

Reply:
People like to copy this file into their non Django related files. I'd like to avoid importing Django here for that reason

Reply:
Could be a plain `warnings.warn`

Also not sure that the 'people like to copy this file' argument holds for Django's codebase. There's `pip install cached-property` for an implementation of this outside of Django.